bundles/org.openhab.binding.hive/src/main/java/org/openhab/binding/hive/internal/client/NodeName.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hive.internal.client;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Represents a Hive API node name.
+ *
+ * @author Ross Brown - Initial contribution
+ */
+@NonNullByDefault
+public final class NodeName extends SimpleValueTypeBase<String> {
```

Comment:
Probably not.  Again I was just using this project to play around and decided to have a go at giving everything strict types.

Reply:
Please remove the strict types since not only do they bloat the code but they also prevent you from retyping some classes that would be much better represented as enums.

Reply:
Things like `NodeName` and `Username` could easily be replaced with strings.

I'm not sure about `UserId`, `NodeId`. These could both be replaced with UUIDs but it seems kind of nice making them kind of opaque objects that you get from the client and shouldn't be creating yourself. What do you think?

I assume the you are talking about `ProductType`, `Protocol` etc. when you mention enums.
I made these pseudo-enum classes because I don't have a full API spec so I don't know all the possible values, just the ones I have seen through my interactions with the API.
I could refactor them so on the public side of the repositories I have enums (including an 'unknown'/'unsupported' value) and on the internal/DTO side I use Strings. Does that sound good to you?

Reply:
> I'm not sure about UserId, NodeId. These could both be replaced with UUIDs but it seems kind of nice making them kind of opaque objects that you get from the client and shouldn't be creating yourself. What do you think?

Well, having made UUID wrapper classes in my own projects I guess it would depend on how you use them. If you are planning to use the UUIDs as keys in maps then I would lean towards using wrappers, otherwise just use the UUIDs as is.

Reply:
I've just pushed a commit that refactors most of these wrapper types @cpmeister 

These are the ones I kept:

- `NodeId` as I do use it as a generic parameter e.g. `DefaultHiveAccountHandler` (line 69):
```java
private final Map<NodeId, HiveThingManager> hiveThingManagers = new HashMap<>();
```

- `UserId` is not used for anything interesting yet but I kept it for consistency with `NodeId`.

- `SessionId` because the string it wraps is actually a JSON Web Token. I may promote it to a "real" type at some point so I can do things like check the session expiry timestamp.

These were converted to enums:
- `ActionType`
- `AttributeName`
- `FeatureType`
- `GroupId`
- `NodeType`
- `ProductType`
- `Protocol`
- `ScheduleType`

These were removed:
- MediaType
- NodeName
- Username
- Password

Does this look good to you now?